protocol/osrs-225/osrs-225-api/src/main/kotlin/net/rsprot/protocol/api/traffic/GenericNetworkTrafficWriter.kt

```diff
@@ -0,0 +1,218 @@
+package net.rsprot.protocol.api.traffic
+
+import net.rsprot.protocol.loginprot.incoming.util.HostPlatformStats
+import net.rsprot.protocol.loginprot.incoming.util.LoginBlock
+import net.rsprot.protocol.metrics.channel.snapshots.impl.GenericChannelTrafficSnapshot
+import net.rsprot.protocol.metrics.channel.snapshots.util.PacketSnapshot
+import net.rsprot.protocol.metrics.snapshots.impl.GenericNetworkTrafficSnapshot
+import net.rsprot.protocol.metrics.writer.NetworkTrafficWriter
+import java.net.InetAddress
+import java.text.NumberFormat
+import java.time.ZoneId
+import java.time.format.DateTimeFormatter
+import java.time.format.FormatStyle
+import kotlin.collections.iterator
+
+public data object GenericNetworkTrafficWriter : NetworkTrafficWriter<GenericNetworkTrafficSnapshot<*>, String> {
```

Comment:
i feel like there's a good case for building another layer on top of this where consumers can set up a regular writing of traffic snapshot contents to a named file in some folder

also perhaps this is better named as a `StringNetworkTrafficWriter`, where there'd also be a `FileNetworkTrafficWriter` that delegates the formatting to this, but has that file-writing-specific code encapsulated within it

what to you think about extending the api a bit for this type of thing? i can give that a crack if you think it'd be valuable. just conscious that all your consumers might end up writing the same thing across the diff projects

Reply:
This is the bit I'm not really sure about in general. There's a lot of logging happening and I feel like most people will not want like half of it, especially when it comes to all the details within the login block.
My thought process here was to make a default implementation that servers can use to get a rough idea of what gets tracked and how, but when it came to actually making use of it, they'd write their own.
Perhaps there is a way to make a nicer API for it, I haven't really figured out anything just yet.

If you want, go ahead and take a crack at it. Do so after I'm done with this PR and have merged it into the master branch, though. That will be when I stop actively updating this feature (it's fine to update it after it's gone to master, this won't be published until after I finish all the things I wanted to do for your server anyhow).

Reply:
fair nuff - i guess it's better to see how this pans out in practice, and make changes that solve real problems. im just making stuff up atm

sounds good to me!